ksqldb-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4

```diff
@@ -153,26 +153,14 @@ windowUnit
     ;
 
 groupBy
-    : groupingElement (',' groupingElement)*
-    ;
-
-groupingElement
-    : groupingExpressions                                               #singleGroupingSet
-    ;
-
-groupingExpressions
-    : '(' (valueExpression (',' valueExpression)*)? ')'
-    | valueExpression
```

Comment:
instead of removing this from our grammar, why not throw in the analyzing phase that "KSQL does not support multiple grouping elements" or whatever. That way the error message is much more understandable

Reply:
Taking your approach to its logical conclusion our grammar should support the full SQL spec and all the validation should be done in the analyzer.

Surely our grammar should reflect our capability?  

Also, I don't even think the syntax I've removed for grouping sets in standard sql.  Standard sequel would look like:

```sql
SELECT
    c1,
    c2,
    aggregate (c3)
FROM
    table
GROUP BY
    GROUPING SETS (
        (c1, c2),
        (c1),
        (c2),
        ()
);
```

Where as the syntax I've removed is:

```sql
SELECT
    c1,
    c2,
    aggregate (c3)
FROM
    table
GROUP BY
    (
        (c1, c2),
        (c1),
        (c2),
        ()
    );
```

Hence my view would be we should remove this from our grammar.   If/when we implement grouping sets we can decide what the correct grammar should be.

Reply:
> Taking your approach to its logical conclusion our grammar should support the full SQL spec and all the validation should be done in the analyzer.

To be honest, I think I would prefer that but it's obviously not a practical amount of work to do now. I get very confused whenever I see a parsing error and I've been wondering how to make those errors better.

> The GROUPING SETS syntax looks to be common across all SQL variants that I've looked at: postgres, snowflake, oracle, mysql, etc

Well that settles it then. LGTM

Reply:
> To be honest, I think I would prefer that but it's obviously not a practical amount of work to do now. I get very confused whenever I see a parsing error and I've been wondering how to make those errors better

I'm actually in agreement here: the antlr parsing errors aren't great.  But, as you say, it's not practical to handle all errors in the engine.